cmd/permissions-checker/permissionschecks.go

```diff
@@ -0,0 +1,130 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os/user"
+	"path/filepath"
+	"regexp"
+	"strings"
+	"syscall"
+)
+
+type currentPermissions struct {
+	permissions           string
+	defaultPermissions    string
+	owner                 string
+	group                 string
+	hasCorrectPermissions bool
+	hasCorrectOwner       bool
+	hasCorrectGroup       bool
+}
+
+// FindRecursive iterates in a directory showing permissions in a recursive way
+/* path [string] 											- where to look for files and directories
+ * defaultPerm [defaultPermissions] 	- data regarding default permissions
+ * 	defaultPerm.file [string]					- default permissions for files
+ * 	defaultPerm.dir [string] 					- default permissions for directories
+ * 	defaultPerm.owner [string] 				- default owner
+ * 	defaultPerm.group [string] 				- default group
+ * search [searchSettings] 						- data regarding search options
+ * 	search.hidden [bool] 							- includes hidden files and directories in the search
+ * 	search.exclude [string] 					- files and/or directories to be excluded (regexp)
+ * 	search.baseDirectory [string]			- base directory
+ * level [int]                        - used to print the output in a hierarchical way
+ */
+func FindRecursive(path string, defaultPerm defaultPermissions, search searchSettings, verbose bool, level int) {
+	files, err := ioutil.ReadDir(path)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// In verbose mode, we are printing the output in a hierarchical way
+	if verbose {
+		level++
+	}
+
+	for _, f := range files {
+		name := f.Name()
+
+		relativePath := strings.TrimPrefix(strings.Join([]string{path, name}, "/"), search.baseDirectory)
+		fullPath := strings.Join([]string{path, name}, "/")
+
+		if m, _ := regexp.MatchString(search.exclude, relativePath); m {
```

Comment:
handle regexp errors.

Probably the search.exclude regexp should be compiled during flag parsing in main, and here we should receive the compiled `*regep.Regexp` structure.